### ISSUE TITLE
Add OpenSUSE support in BinTray install.sh

### ISSUE THREAD

**Issue by RespiteSage** [crystal-lang/distribution-scripts#74]
The new BinTray `install.sh` script fails on OpenSUSE Tumbleweed 20200817 (and, I suspect, most OpenSUSE distributions).

In `_discover_distro_type`, execution continues until `if hash apt-get 2>/dev/null`, which evaluates as true and sets `DISTRO_TYPE` to `"deb"`, when, by default, OpenSUSE uses Zypper, which is RPM-based. Actually, the reason for that `if` evaluating to true is that, for some reason, `apt-get` "forwards" to Zypper by default on OpenSUSE instead of just not existing. `DISTRO_TYPE` equal to `"deb"` later causes `apt-key` to fail because there is no such command.

That is, of course, only the first issue. The script as it stands seems to support only Yum and Apt, whereas OpenSUSE support would likely require the use of Zypper. I hope to put in a PR for this myself, but at least this issue will exist if I don't get around to it.

**Comment by j8r** [comment]
OpenSuse has an `apt-get` symlink to `zypper`?! What about `dpkg`, could do the trick if it does not symlink to `rpm`.

By the way, just check for `zypper` at first, and it will be also good.

**Comment by RespiteSage** [comment]
> OpenSuse has an `apt-get` symlink to `zypper`?!
Actually, it's more complicated than this... `apt-get` in `/usr/bin/` is symlinked to `/usr/bin/aptitude`, which is a perl script that appears to convert `apt` arguments to `zypper` arguments (e.g. `--with-recommends` becomes `--recommends`) and then calls `zypper` with those arguments.

**Comment by aboe76** [comment]
I think to have suse support the install script needs to be adjusted as follows:
add 
 `_match_etc_issue "rpm" "Suse" && return` in the `discover_distro_type()` function

and create a new exception based on `DISTRO=Suse` to have this repo file created:
```
crystal.repo
[crystal]
name=crystal
enabled=1
autorefresh=1
baseurl=https://dl.bintray.com/crystal/rpm/all/x86_64/stable/
type=rpm-md
priority=99
gpgcheck=1
gpgkey=http://bintray.com/user/downloadSubjectPublicKey?username=bintray
keeppackages=0
```
and at last the install command for `Suse` should be:
```
zypper install -y --gpg-auto-import-keys $CRYSTAL_PACKAGE
```

**Comment by straight-shoota** [comment]
This has been resolved by #91. Packages for openSUSE are now available through OBS (https://crystal-lang.org/install/on_opensuse/).
